### ISSUE TITLE
Use cargo-xcode instead of cargo-lipo for XCode setup tutorial

### ISSUE THREAD

**Issue by antoniusnaumann** [chinedufn/swift-bridge#112]
Since [cargo-lipo is considered deprecated by its developer](https://github.com/TimNN/cargo-lipo#maintenance-status), the [tutorial for setting up swift-bridge in an xcode project](https://chinedufn.github.io/swift-bridge/building/xcode-and-cargo/index.html) should use a properly maintained alternative instead.
[cargo-xcode](https://crates.io/crates/cargo-xcode) seems to be a promising alternative since it also does some of the required XCode project configuration and therefore eliminates the need for additional setup/build scripts (build.rs is still necessary but minimal).

I was able to set up and built a project using swift-bridge + cargo-xcode with the Apple Multiplatform target, so I could do a draft for the revised tutorial section if wanted.

**Comment by chinedufn** [comment]
Hey!

Is there any reason not to just use `lipo` directly?

i.e. something like: https://github.com/chinedufn/swift-bridge/blob/89dc1585c51063a4bc95a0a77319dc903c6b349c/SwiftRustIntegrationTestRunner/swift-package-rust-library-fixture/build.sh#L13-L16

**Comment by antoniusnaumann** [comment]
Mostly convenience regarding the XCode setup and not having to maintain an additional shell script.

Small difference overall, but I think that would make it less intimidating for beginners.

**Comment by chinedufn** [comment]
I see the following in the README

> Rust binaries are exported as command-line tools. This tool intentionally does not make app bundles. If you want to build a Mac GUI app, create one as ObjC or Swift project in Xcode and run Rust code from a Rust static library.

Does this mean that `cargo xcode` does not work for iOS and macOS applications that have GUIs?

If so, then we'd have to teach people two ways of setting up their application. Not ideal.

---

I definitely like your goal of simplifying things and removing steps. Just trying to get a good sense of if `cargo xcode` works across all use cases.

If it doesn't, are there other ways that we can simplify our tutorial?

**Comment by antoniusnaumann** [comment]
@chinedufn, no, it works in GUI-Apps. I embedded a Rust library using cargo xcode in a SwiftUI app that targets both macOS and iOS.

What this line means: You cannot use cargo xcode to bundle pure Rust binaries for macOS and iOS, but this irrelevant for swift-bridge.

**Comment by chinedufn** [comment]
Got it, thanks!

Whether or not the beginner is using `cargo-xcode` or `lipo`, they're going to be copy-pasting things that they've probably never seen before.

`cargo-xcode` has the pro of saving a script, but the con of being an additional thing to install.

It doesn't seem to me like a lot is being saved by introducing `cargo-xcode`. 

So, I'd rather teach people to use `lipo` directory. 
It's much more likely to be around in a decade since it's maintained by Apple, comes with your OS and isn't limited to just cargo based builds. 

In general, I would prefer for users to use `lipo` instead of `cargo-xcode`, unless `cargo-xcode` saved a ton of headaches, which it doesn't look it it would.

All of that being said, I haven't actually used `cargo-xcode`, so let me know if I'm wrong.

Thank you again for explaining!
